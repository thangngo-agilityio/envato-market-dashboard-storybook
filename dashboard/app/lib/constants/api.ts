// TODO: Will provide latter
export const BASE_API = '';
export const USERS_API = process.env.NEXT_PUBLIC_API_USER || BASE_API;
export const STATISTICAL_API =
  process.env.NEXT_PUBLIC_STATISTICAL_API || BASE_API;

export const END_POINTS = {
  AUTHEN_USERS: '/authenUsers',
  USERS: '/users',
  TRANSACTIONS: '/transactions',
  PRODUCTS: '/products',
  RECENT_ACTIVITIES: '/activities',
  STATISTICS: '/statistics',
  REVENUE: '/revenue',
  EFFICIENCY: '/efficiency',
  NOTIFICATION: '/notifications',
  EMPLOYEES: '/employees',
  OVERALL_BALANCE: '/overall-balance',
  SIGN_IN: '/sign-in',
  SIGN_UP: '/sign-up',
  DELETE_TRANSACTION: '/delete-transactions',
  EDIT_TRANSACTION: '/edit-transactions',
  CREATE_PIN: '/create-pin-code',
  CONFIRM_PIN: '/pin-code',
  UPDATE_PASSWORD: '/update-password',
  MY_WALLET: '/wallets',
  ADD_MONEY: '/add-money',
  SEND_MONEY: '/send-money',
  SUPPORT: '/supports',
  LOCK: '/lock',
  UNLOCK: '/unlock',
  ADMIN: '/admin',
  SETTINGS: '/settings',
  LOGIN: '/login',
  WALLET: '/my-wallets',
};

export const SEARCH_PARAM = {
  EMAIL: 'email',
  PASSWORD: 'password',
  TRANSACTION_NAME: 'transactionName',
};
