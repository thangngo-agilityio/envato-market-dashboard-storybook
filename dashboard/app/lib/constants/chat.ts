export const FIREBASE_CHAT = {
  USERS: 'users',
  USER_CHATS: 'userChats',
  CHATS: 'chats',
};

export const USER_CHATS_FIELD = {
  USER_INFO: '.userInfo',
  DATE: '.date',
  LAST_MESSAGE: '.lastMessage',
};
