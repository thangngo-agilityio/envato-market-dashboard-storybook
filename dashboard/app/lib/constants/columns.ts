// Types
import {
  TDataSource,
  TProduct,
  TProductResponse,
  TRecentActivities,
  TTransaction,
} from '../interfaces';

export const COLUMNS_DASHBOARD = (
  onRenderHead: (title: string, key: string) => void,
  onRenderRole: (role: TTransaction) => void,
  onRenderBody: ({ id, image, name }: TDataSource) => void,
  onRenderActionIcon: (data: TTransaction) => void,
  onRenderSpent: (spent: TTransaction) => void,
  onRenderEmail: (email: TTransaction) => void,
) => [
  {
    title: 'Customer name',
    key: 'name',
    renderHead: onRenderHead,
    renderBody: onRenderBody,
  },
  {
    title: 'Email',
    key: 'email',
    renderHead: onRenderHead,
    renderBody: onRenderEmail,
  },
  {
    title: 'Location',
    key: 'location',
    renderHead: onRenderHead,
  },
  {
    title: 'Spent',
    key: 'spent',
    renderHead: onRenderHead,
    renderBody: onRenderSpent,
  },
  {
    title: 'Role',
    key: 'role',
    renderHead: onRenderHead,
    renderBody: onRenderRole,
  },
  {
    title: '',
    key: 'action',
    renderBody: onRenderActionIcon,
    renderHead: onRenderHead,
  },
];

export const COLUMNS_HISTORY = (
  renderHead: (title: string, key: string) => void,
  renderBody: ({ id, image, name }: TDataSource) => void,
  renderPaymentStatus: ({ paymentStatus }: TDataSource) => void,
  renderTransactionStatus: ({ paymentStatus }: TDataSource) => void,
  renderSpent: (spent: TTransaction) => void,
) => [
  {
    title: 'Customer name',
    key: 'name',
    renderHead: renderHead,
    renderBody: renderBody,
  },
  {
    title: 'Date',
    key: 'date',
    renderHead: renderHead,
  },
  {
    title: 'Amount',
    key: 'spent',
    renderHead: renderHead,
    renderBody: renderSpent,
  },
  {
    title: 'Payment',
    key: 'paymentStatus',
    renderHead: renderHead,
    renderBody: renderPaymentStatus,
  },
  {
    title: 'Status',
    key: 'transactionStatus',
    renderHead: renderHead,
    renderBody: renderTransactionStatus,
  },
];

export const COLUMNS_PRODUCTS = (
  onRenderHead: (title: string, key: string) => void,
  onRenderBody: ({ id, name }: TDataSource) => void,
  onRenderGallery: ({ imageURLs, name }: TProduct) => void,
  onRenderPrice: (amount: TProduct) => void,
  onRenderStatus: (productStatus: TDataSource) => void,
  onRenderQuantity: (stock: TProduct) => void,
  onRenderActionIcon: (data: TProductResponse) => void,
) => [
  {
    title: 'Product Name',
    key: 'name',
    renderHead: onRenderHead,
    renderBody: onRenderBody,
  },
  {
    title: 'Gallery Thumbnail',
    key: 'imageURLs',
    renderHead: onRenderHead,
    renderBody: onRenderGallery,
  },
  {
    title: 'Price',
    key: 'price',
    renderHead: onRenderHead,
    renderBody: onRenderPrice,
  },
  {
    title: 'Status',
    key: 'productStatus',
    renderHead: onRenderHead,
    renderBody: onRenderStatus,
  },
  {
    title: 'Quantity',
    key: 'quantity',
    renderHead: onRenderHead,
    renderBody: onRenderQuantity,
  },
  {
    title: 'Date',
    key: 'date',
    renderHead: onRenderHead,
  },
  {
    title: '',
    key: 'action',
    renderBody: onRenderActionIcon,
    renderHead: onRenderHead,
  },
];

export const COLUMNS_RECENT_ACTIVITIES = (
  onRenderHead: (title: string, key: string) => void,
  onRenderBody: ({ id, actionName }: TDataSource) => void,
  onRenderName: (actionName: TRecentActivities) => void,
  onRenderEmail: (email: TRecentActivities) => void,
) => [
  {
    title: 'ID Action',
    key: '_id',
    renderHead: onRenderHead,
    renderBody: onRenderBody,
  },
  {
    title: 'Action Name',
    key: 'actionName',
    renderHead: onRenderHead,
    renderBody: onRenderName,
  },
  {
    title: 'Email',
    key: 'email',
    renderHead: onRenderHead,
    renderBody: onRenderEmail,
  },
  {
    title: 'Date',
    key: 'date',
    renderHead: onRenderHead,
  },
];
