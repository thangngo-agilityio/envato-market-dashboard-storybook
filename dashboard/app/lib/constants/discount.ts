export const DEFAULT_DISCOUNT_PERCENTAGE = 0.6;
export const LIMIT_OF_BONUS = 5;
