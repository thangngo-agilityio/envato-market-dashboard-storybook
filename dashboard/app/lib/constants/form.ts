import { ERROR_MESSAGES } from './messages';
import { REGEX } from './regex';

export const AUTH_SCHEMA = {
  FIRST_NAME: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('First Name'),
  },
  LAST_NAME: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Last Name'),
  },
  STREET: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Street'),
  },
  STATE: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('State'),
  },
  CITY: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('City'),
  },
  EMAIL: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Email'),
    pattern: {
      value: REGEX.EMAIL,
      message: ERROR_MESSAGES.EMAIL_INVALID,
    },
  },
  PASSWORD: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Password'),
  },
  CONFIRM_PASSWORD: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Confirm password'),
    validate: (val: string, { password }: { password: string }) => {
      if (password && val !== password) {
        return ERROR_MESSAGES.PASSWORD_NOT_MATCH;
      }
    },
  },
  REMEMBER_ME: {
    required: false,
  },
  AGREE_POLICY: {
    validate: (
      value: boolean,
      //eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unused-vars
      { isAcceptPrivacyPolicy: __, ...fieldValues }: any,
    ) => Object.values(fieldValues).every((value) => value) && value,
  },

  PHONE_NUMBER: {
    pattern: {
      value: REGEX.PHONE_NUMBER,
      message: ERROR_MESSAGES.PHONE_NUMBER_INVALID,
    },
  },

  AVATAR_URL: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Avatar'),
    pattern: {
      value: REGEX.IMG,
      message: ERROR_MESSAGES.UPLOAD_IMAGE,
    },
  },

  FACEBOOK: {
    pattern: {
      value: REGEX.URL,
      message: ERROR_MESSAGES.FACEBOOK_INVALID,
    },
  },

  TWITTER: {
    pattern: {
      value: REGEX.URL,
      message: ERROR_MESSAGES.TWITTER_INVALID,
    },
  },

  LINKEDIN: {
    pattern: {
      value: REGEX.URL,
      message: ERROR_MESSAGES.LINKEDIN_INVALID,
    },
  },

  YOUTUBE: {
    pattern: {
      value: REGEX.URL,
      message: ERROR_MESSAGES.YOUTUBE_INVALID,
    },
  },

  ZIP_CODE: {
    pattern: {
      value: REGEX.ZIP_CODE,
      message: ERROR_MESSAGES.ZIP_CODE_INVALID,
    },
  },

  PIN_CODE: {
    required: true,
    validate: (value: string) => value.length === 4,
  },

  TRANSFER_AMOUNT: {
    required: true,
    validate: (value: string) => !!value.length && !isNaN(+value),
  },
  OLD_PASSWORD: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Old password'),
  },
  NEW_PASSWORD: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('New password'),
    validate: (value: string) => {
      if (!REGEX.LENGTH_IS_EIGHT.test(value)) {
        return ERROR_MESSAGES.PASS_WORD_SHORT;
      }

      if (!REGEX.PASSWORD.test(value)) {
        return ERROR_MESSAGES.PASS_WORD_WEAK;
      }

      return true;
    },
  },

  NAME: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Name'),
  },

  AMOUNT: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Amount'),
    validate: (value: string) => {
      if (Number(value) <= 0) {
        return ERROR_MESSAGES.AMOUNT_INVALID;
      }
      if (!REGEX.LIMIT_NUMBER.test(value)) {
        return ERROR_MESSAGES.LIMIT_AMOUNT;
      }

      return true;
    },
  },

  QUANTITY: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Quantity'),
    validate: (value: string) => {
      if (Number(value) < 0) {
        return ERROR_MESSAGES.QUANTITY_INVALID;
      }
      if (!REGEX.LIMIT_NUMBER.test(value)) {
        return ERROR_MESSAGES.LIMIT_QUANTITY;
      }

      return true;
    },
  },

  GALLERY_THUMBNAIL: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Gallery Thumbnail'),
  },
};
