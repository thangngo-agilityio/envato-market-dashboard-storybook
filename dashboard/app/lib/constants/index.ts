export * from './routers';
export * from './time';
export * from './images';
export * from './form';
export * from './messages';
export * from './regex';
export * from './texts';
export * from './themes';
export * from './api';
export * from './images';
export * from './menu';
export * from './sidebar';
export * from './charts';
export * from './sidebar';
export * from './common';
export * from './options';
export * from './transaction';
export * from './store';
export * from './status';
export * from './pagination';
export * from './columns';
export * from './faq';
export * from './avatar';
export * from './sizes';
export * from './setting';
export * from './termAndCondition';
export * from './notification';
export * from './authentication';
export * from './discount';
export * from './metadata';
export * from './og-properties';
export * from './chat';
export * from './product';
