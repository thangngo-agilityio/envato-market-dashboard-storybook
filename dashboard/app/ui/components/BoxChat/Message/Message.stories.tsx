import { StoryObj, Meta } from '@storybook/react';

// Constants
import { IMAGES } from '@/lib/constants';

// Components
import Message from '.';

const meta: Meta<typeof Message> = {
  title: 'Custom Components/Message',
  component: Message,
  tags: ['autodocs'],
  argTypes: {
    content: {
      description: 'The message content',
    },

    avatarUser: {
      description: 'The URL of the message avatar ',
    },

    avatarAdmin: {
      description: 'The URL of the message avatar ',
    },

    localeTime: {
      description: 'Indicate the time of the message',
    },
  },
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;
type Story = StoryObj<typeof Message>;

export const Default: Story = {
  args: {
    content: 'This is message',
    avatarAdmin: IMAGES.AVATAR_SIGN_UP.url,
    avatarUser: IMAGES.AVATAR_SIGN_UP.url,
  },
};
