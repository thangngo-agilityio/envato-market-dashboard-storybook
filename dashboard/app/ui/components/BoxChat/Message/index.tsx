import { Box, Flex, Spacer, Text } from '@chakra-ui/react';
import { useEffect, useState } from 'react';
import Image from 'next/image';

// Constants
import { IMAGES } from '@/lib/constants';

// Utils
import { generatePlaceholder } from '@/lib/utils';

interface MessageProps {
  content?: string;
  avatarUser?: string;
  avatarAdmin?: string;
  localeTime?: string;
  senderId: string;
  superAdminId: string;
}

const Message = ({
  senderId,
  content = '',
  avatarUser = IMAGES.CHAT_USER_AVATAR.url,
  avatarAdmin = IMAGES.CHAT_USER_AVATAR.url,
  localeTime = new Date().toLocaleTimeString([], {
    hour: 'numeric',
    minute: '2-digit',
  }),
  superAdminId,
}: MessageProps) => {
  const [adminId, setAdminId] = useState<string>('');

  useEffect(() => {
    const fetchData = async () => {
      setAdminId(superAdminId || '');
    };

    fetchData();
  }, []);

  const isAdmin = senderId === adminId;

  return (
    <Flex width="100%" justifyContent={!isAdmin ? 'flex-end' : 'flex-start'}>
      {isAdmin && (
        <Box pos="relative" w={9} h={9}>
          <Image
            src={avatarAdmin}
            alt={avatarAdmin}
            fill
            sizes="100vw"
            placeholder="blur"
            blurDataURL={generatePlaceholder(36, 36)}
            style={{
              borderRadius: '50%',
              marginRight: '8px',
            }}
          />
        </Box>
      )}

      <Flex
        align="flex-end"
        direction={!isAdmin ? 'row-reverse' : 'row'}
        mb="30px"
        alignItems="center"
      >
        <Box
          data-testid="image-container"
          bg={!isAdmin ? 'primary.300' : 'background.section.messageUser'}
          p={3}
          ml={2}
          borderRadius={8}
          color="text.primary"
          dangerouslySetInnerHTML={{ __html: content ?? '' }}
        />
        <Spacer />

        <Text
          ml={3}
          mr={3}
          fontSize="xs"
          color="text.textTime"
          fontWeight="medium"
          minW="max-content"
        >
          {localeTime}
        </Text>
      </Flex>

      {!isAdmin && (
        <Box pos="relative" w={9} h={9}>
          <Image
            src={avatarUser}
            alt={avatarUser}
            fill
            sizes="100vw"
            placeholder="blur"
            blurDataURL={generatePlaceholder(36, 36)}
            style={{
              borderRadius: '50%',
              marginLeft: '8px',
            }}
          />
        </Box>
      )}
    </Flex>
  );
};
export default Message;
