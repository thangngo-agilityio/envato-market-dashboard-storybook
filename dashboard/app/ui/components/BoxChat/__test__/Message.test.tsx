import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Constants
import { IMAGES } from '@/lib/constants';

// Message
import Message from '@/ui/components/BoxChat/Message';

describe('Message component', () => {
  const mockLocaleTime = new Date(1702543868252).toLocaleTimeString([], {
    hour: 'numeric',
    minute: '2-digit',
  });

  it('renders correctly', () => {
    const { container } = render(
      <Message
        content="This is message"
        avatarAdmin={IMAGES.CHAT_USER_AVATAR.url}
        avatarUser={IMAGES.CHAT_USER_AVATAR.url}
        localeTime={mockLocaleTime}
        senderId="1"
        superAdminId="1"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render text content and current time when no input localeTime and isImage parameters', () => {
    const { getByTestId } = render(
      <Message
        senderId="1"
        superAdminId="1"
        content="Hello"
        avatarAdmin={IMAGES.CHAT_USER_AVATAR.url}
        avatarUser={IMAGES.CHAT_USER_AVATAR.url}
      />,
    );

    const textContent = getByTestId('image-container');
    expect(textContent).toBeInTheDocument();
  });

  it('should render avatar after the content when avatarPosition is "after"', () => {
    const { getByTestId } = render(
      <Message
        senderId="1"
        superAdminId="1"
        content="Hello"
        avatarAdmin={IMAGES.CHAT_USER_AVATAR.url}
        avatarUser={IMAGES.CHAT_USER_AVATAR.url}
        localeTime={mockLocaleTime}
      />,
    );

    const avatar = getByTestId('image-container');
    expect(avatar).toBeInTheDocument();
    expect(avatar).toHaveStyle('margin-left: 2px;');
  });
});
