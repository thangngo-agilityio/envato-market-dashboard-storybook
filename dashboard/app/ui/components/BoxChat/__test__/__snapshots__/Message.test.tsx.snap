// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders correctly 1`] = `
<div>
  <div
    class="css-15rwwo"
  >
    <div
      class="css-1uzq3xz"
    >
      <img
        alt="/images/avatar-user-chat.webp"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar-user-chat.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; border-radius: 50%; margin-right: 8px; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      />
    </div>
    <div
      class="css-cr18eb"
    >
      <div
        class="css-1nsc086"
        data-testid="image-container"
      >
        This is message
      </div>
      <div
        class="css-17xejub"
      />
      <p
        class="chakra-text css-haer8f"
      >
        3:51 PM
      </p>
    </div>
  </div>
</div>
`;
